keystrokes/src/components/Character/CharacterList.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState, useEffect } from 'react';
+import '../../stylesheets/character.css';
+import styleGuide from '../../stylesheets/style';
+import { ref } from '../../config/constants';
+import { List, ListItem, Typography, Divider } from '@material-ui/core';
+import CharacterCreationModal from './CharacterCreationModal';
+
+function CharacterList(props) {
+	const [characters, setCharacters] = useState({});
+	const [currentCharacter, setCurrentCharacter] = useState()
+	const [open, setOpen] = useState(false);
+	const classes = styleGuide();
+
+	useEffect(() => {
+		console.log(props)
+		setCharacters(props.userInfo.characters || {});
+		setCurrentCharacter(props.userInfo.currentCharacter);
+	}, [props]);
+
+	const handleOpen = () => {
+		setOpen(true);
+	}
+
+	async function handleSubmit(fields) {
+		const {name, characterClass, spec, race} = fields
+		let charactersSnapShot = await ref.child("/Characters");
+		let doesCharacterExist = await charactersSnapShot.child(name).once('value');
+		if (doesCharacterExist.exists()) {
+			console.log("error")
+			//throw error since name is currently id - replace with real key eventually
+			return;
+		} else {
+			charactersSnapShot
+			.child(name).set({
+				"class": characterClass, 
+				"race": race, 
+				"name": name, 
+				"specs": {}
+			});
+			charactersSnapShot
+			.child(name)
+			.child("specs")
+			.child(spec)
+			.set({
+				"spec": spec, 
+				"selected": true,
+				"configured": false
+			});
+		}
+		let snapshot = await ref.child(props.userPath);
+		let userCharacterData = await snapshot.once('value');
```

Comment:
since this is kinda like global state for the application we might want to get it higher up and make it realtime so when changes come in they automatically update. it'd save you from asking for this data again and seeing if it exists

Reply:
so, user path is point at the users, and the one above it is actually pointing at the characters, which are two seperate tables. Maybe not the best design choice... i think i had watched a video about using duplication to avoid joins and to make queries fast.... this might be a case of premature optimization though haha. I might end up changing the db design up, now that im actually using it. dbTestData is a mock of how the data will look in the end (as of now)